src/snowflake/snowpark/_internal/compiler/utils.py

```diff
@@ -323,3 +327,75 @@ def get_snowflake_plan_queries(
 def is_active_transaction(session):
     """Check is the session has an active transaction."""
     return session._run_query("SELECT CURRENT_TRANSACTION()")[0][0] is not None
+
+
+def plot_plan_if_enabled(root: TreeNode, filename: str) -> None:
```

Comment:
Can we add a test for this?

Reply:
added a test to ensure we are behaving correctly based on param and expected files are present.

Reply:
Exactly, that's what I had in mind. Thanks!